xc/xenops_server_xen.ml

```diff
@@ -540,18 +540,25 @@ let device_by_id xc xs vm kind domain_selection id =
     raise (Does_not_exist("domain", vm))
   | Some frontend_domid ->
     let fetch_all_from_xenstore_f () = (* expensive *)
-      let devices = Device_common.list_frontends ~xs frontend_domid in
+      let devices = (Device_common.list_frontends ~xs frontend_domid)
+        @ (Device_common.list_xenserver_devices ~xs frontend_domid)
+      in
       let key = _device_id kind in
       let id_of_device device =
-        let path = Device_common.get_private_data_path_of_device device in
+        let path = match kind with
+          | Device_common.NetSriovVf -> Device_common.extra_xenserver_path_of_device ~xs device
+          | _ -> Device_common.get_private_data_path_of_device device
```

Comment:
I didn't like the catch all `_` here but the number of cases to match is substantial, so it is probably not worth to be explicit